src/tools/perf/cuda/ucp_cuda_kernel.cu

```diff
@@ -95,16 +101,37 @@ public:
         return m_pending_count;
     }
 
+    template<ucs_device_level_t level>
+    __device__ inline unsigned get_channel_id() const
+    {
+        switch (m_channel_mode) {
+        case UCX_PERF_CHANNEL_MODE_SINGLE:
+            return 0;
+        case UCX_PERF_CHANNEL_MODE_RANDOM:
+            return curand(m_rand_state) % m_num_channels;
```

Comment:
maybe

```suggestion
            return curand(m_rand_state) % (gridDim.x * blockDim.x);
```

Reply:
do we really need to save num_channels?
can we calculate `ucx_perf_cuda_thread_index<level>(gridDim.x * blockDim.x)` ?